src/sentry/static/sentry/app/views/settings/components/forms/projectMapperField.tsx

```diff
@@ -49,36 +59,78 @@ export class RenderField extends React.Component<RenderProps> {
       item => !mappedValuesUsed.has(item.value)
     );
 
+    const handleSelectProject = ({value}: {value: number}) => {
+      this.setState({selectedSentryProjectId: value});
+    };
+
+    const handleSelectMappedValue = ({value}: {value: MappedValue}) => {
+      this.setState({selectedMappedValue: value});
+    };
+
     const handleAdd = () => {
-      const {value: sentryProjectId} = this.sentryProjectRef.current.state.value;
-      const {value: mappedValue} = this.mappedRef.current.state.value;
       //add the new value to the list of existing values
-      const projectMappings = [...existingValues, [sentryProjectId, mappedValue]];
+      const projectMappings = [
+        ...existingValues,
+        [selectedSentryProjectId, selectedMappedValue],
+      ];
       //trigger events so we save the value and show the check mark
       onChange?.(projectMappings, []);
       onBlur?.(projectMappings, []);
+      this.setState({selectedSentryProjectId: null, selectedMappedValue: null});
     };
 
-    const renderItem = (itemTuple: [number, any]) => {
+    const handleDelete = (index: number) => {
+      const projectMappings = existingValues
+        .slice(0, index)
+        .concat(existingValues.slice(index + 1));
+      //trigger events so we save the value and show the check mark
+      onChange?.(projectMappings, []);
+      onBlur?.(projectMappings, []);
+    };
+
+    const renderItem = (itemTuple: [number, any], index: number) => {
       const [projectId, mappedValue] = itemTuple;
-      const {slug} = sentryProjectsById[projectId];
+      const project = sentryProjectsById[projectId];
       // TODO: add special formatting if deleted
-      const {label: itemLabel} = mappedItemsByValue[mappedValue] || {label: 'Deleted'};
+      const mappedItem = mappedItemsByValue[mappedValue];
       return (
-        <Item key={projectId}>
-          <ItemValue>{slug}</ItemValue> <ItemValue>{itemLabel}</ItemValue>
+        <Item key={index}>
+          {project ? (
+            <StyledIdBadge
+              project={project}
+              avatarSize={20}
+              displayName={project.slug}
+              avatarProps={{consistentWidth: true}}
+            />
+          ) : (
+            <ItemValue>Deleted</ItemValue>
+          )}
+          <MappedItemValue>
+            {mappedItem ? (
+              <React.Fragment>
+                <StyledVercelIcon />
+                {mappedItem.label}
+                <StyledExternalLink href={mappedItem.url}>
+                  <IconOpen />
+                </StyledExternalLink>
+              </React.Fragment>
+            ) : (
+              'Deleted'
+            )}
+          </MappedItemValue>
+          <DeleteButton
+            onClick={() => handleDelete(index)}
+            icon={<IconDelete color="gray500" />}
+            size="small"
+            type="button"
+          />
```

Comment:
this should have an `aria-label` because there's no button text - e.g. "delete <project.slug>"